### ISSUE TITLE
Adobe XD Files (.xd) Preview

### ISSUE THREAD

**Issue by idesignwebsites** [QL-Win/QuickLook#307]
Please add QuickLook support for the Adobe XD files (.xd). 

Adobe XD is a new app from Adobe for UX/UI designers. It's growing fast and it's quickly replacing Photoshop and Illustrator for these purposes. https://www.adobe.com/uk/products/xd.html 

Despite Adobe XD is an UWP app, Windows Explorer doesn't support XD thumbnails natively.

**Comment by xupefei** [comment]
This is unfortunately not possible yet because there no one can open and render .xd files except Adobe XD itself.
The support from QL side can be added only when there is a library for this format.

**Comment by idesignwebsites** [comment]
Thank you for the reply. I 100% understand. It's actually very sad that Adobe and Microsoft are not commited more to make these things work properly and deliver a better experience to their clients. Thankfully there are some great people like you so I'll use the opportunity to thank you for making the difference.

**Comment by jeremyhart** [comment]
I fully agree that this would be seriously helpful. I have tried to find out from Adobe when we can expect to see an SDK or library to do this, I'll update this issue when I hear more.

**Comment by ColaIan** [comment]
Hello, I see that this issue hadn't been updated for one and half a year, and I'd like to know if it's possible to request for .sketch file preview also in this issue (If not I'll open another plugin request)

**Comment by iamogbz** [comment]
> This is unfortunately not possible yet because there no one can open and render .xd

XD files can be unzipped

```sh
➜  example git:(main) ✗ unzip file01.xd
Archive:  file01.xd
 extracting: mimetype                
  inflating: manifest                
 extracting: META-INF/metadata.xml   
  inflating: artwork/artboard-6ca57bbd-3a31-4844-b3e0-8b2a839bc8ac/graphics/graphicContent.agc  
  inflating: artwork/pasteboard/graphics/graphicContent.agc  
  inflating: interactions/interactions.json  
 extracting: preview.png             
 extracting: renditions/image-420-420.png  
  inflating: resources/graphics/graphicContent.agc  
  inflating: sharing/sharing.json    
 extracting: thumbnail.png
```

If this helps anyone looking to build a plugin, don't have the xcode experience but it should be possible to extract the `preview.png` and `thumbnail.png` file binary and use it for the quicklook preview and thumbnail respectively.

**Comment by emako** [comment]
Any supporting about abobe files, just check https://github.com/QL-Win/QuickLook/issues/1516
